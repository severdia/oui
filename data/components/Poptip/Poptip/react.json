{"description":"","methods":[],"props":{"children":{"type":{"name":"string"},"required":true,"description":"Content that appears within the popover body"},"testSection":{"type":{"name":"string"},"required":false,"description":"Hook for automated JavaScript tests"}},"private":true}