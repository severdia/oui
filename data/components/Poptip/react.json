{"description":"Displays help text when hovering on an element.","displayName":"Poptip","methods":[],"props":{"children":{"type":{"name":"node"},"required":true,"description":"Content that, when hovered on, makes the Poptip appear"},"content":{"type":{"name":"union","value":[{"name":"node"},{"name":"string"}]},"required":false,"description":"content could be a node of a string"},"disable":{"type":{"name":"bool"},"required":false,"description":"disable value | default => false"},"position":{"type":{"name":"enum","value":[{"value":"'top'","computed":false},{"value":"'bottom'","computed":false},{"value":"'left'","computed":false},{"value":"'right'","computed":false}]},"required":false,"description":"position value options | default => top"},"theme":{"type":{"name":"string"},"required":false,"description":"background color | default => dark"},"trigger":{"type":{"name":"enum","value":[{"value":"'mouseenter'","computed":false},{"value":"'focus'","computed":false},{"value":"'click'","computed":false},{"value":"'manual'","computed":false}]},"required":false,"description":"trigger value options | default => mouseenter focus"}},"private":false}