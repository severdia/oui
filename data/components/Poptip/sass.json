{"name":"Poptip\n","description":"Information bubbles on hover.","start":{"line":1,"column":1},"end":{"line":82,"column":4},"example":[{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-top-left\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-top-center\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-top-right\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-right-top\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-right-center\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-right-bottom\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-bottom-right\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-bottom-center\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-bottom-left\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-left-bottom\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-left-center\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-left-top\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"}]}