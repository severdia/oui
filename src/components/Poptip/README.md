# Poptips
## Poptips (a.k.a. Tooltips)

Use poptips to provide additional, non-essential information to users, such as explanatory help text. Poptips should only be 1–2 sentences in length, and should not contain links, buttons, images, interactive elements, or formatted text (titles, bolds, italics, etc.). If you need to provide more than a couple of sentences of content, or want to include links, you should revise your content, or reconsider using a tooltip at all. Alternatively, consider using a Popover.

![alt text](http://design.optimizely.com/assets/img/design-patterns/product/poptips-and-popovers/poptip.png "good example of Poptip component")

## Good Examples

![alt text](http://design.optimizely.com/assets/img/design-patterns/product/poptips-and-popovers/poptip-example-good-help.png "good example of Poptip component")

This poptip uses short text and doesn't contain links or any other elements.

---

![alt text](http://design.optimizely.com/assets/img/design-patterns/product/poptips-and-popovers/poptip-example-good-icon.png "good example of Poptip component")

This poptip is a label for this icon button.

## Bad Example

![alt text](http://design.optimizely.com/assets/img/design-patterns/product/poptips-and-popovers/poptip-example-bad-long.png "bad example of Poptip component usage")

This poptip's content is way too long, and includes styled text and images. You should just link to a Knowledge Base article with this content instead. It’s also styled as a popover even though it appears on hover.

--- 

![alt text](http://design.optimizely.com/assets/img/design-patterns/product/poptips-and-popovers/poptip-example-bad-link.png "bad example of Poptip component usage")

This poptip's length is appropriate, but it should not contain a link.

