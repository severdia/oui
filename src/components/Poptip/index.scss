////
/// Poptip
/// @description Information bubbles on hover.
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-top-left" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-top-center" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-top-right" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-right-top" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-right-center" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-right-bottom" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-bottom-right" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-bottom-center" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-bottom-left" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-left-bottom" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-left-center" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
/// @example[html]
///   <div style="position: relative; min-height: 2rem">
///     <div class="oui-pop--tip oui-pop--tip--arrow-left-top" style="opacity: 1;">
///       This is a pop-tip. Use only limited text.
///     </div>
///   </div>
////

.#{$namespace}pop--tip,
.oui-pop--tip,
.tippy-popper {
  border-radius: var(--border-radius);
}

.#{$namespace}pop--tip,
.oui-pop--tip {
  background: map-fetch($color, ui, dark);
  color: map-fetch($color, text, white);
  font-size: map-fetch($font, size, micro);
  padding: spacer(0.5) spacer(1);
  max-width: map-fetch($poptip, max-width);
  display: inline-block;
  z-index: map-fetch($z-index, poptip);
  font-weight: 500;
  position: absolute;
  opacity: 0;

  // Arrow options.

  &--arrow-top-left {
    @include arrow(top, left, map-fetch($color, ui, dark));
  }

  &--arrow-top-center {
    @include arrow(top, center, map-fetch($color, ui, dark));
  }

  &--arrow-top-right {
    @include arrow(top, right, map-fetch($color, ui, dark));
  }

  &--arrow-right-top {
    @include arrow(right, top, map-fetch($color, ui, dark));
  }

  &--arrow-right-center {
    @include arrow(right, center, map-fetch($color, ui, dark));
  }

  &--arrow-right-bottom {
    @include arrow(right, bottom, map-fetch($color, ui, dark));
  }

  &--arrow-bottom-right {
    @include arrow(bottom, right, map-fetch($color, ui, dark));
  }

  &--arrow-bottom-center {
    @include arrow(bottom, center, map-fetch($color, ui, dark));
  }

  &--arrow-bottom-left {
    @include arrow(bottom, left, map-fetch($color, ui, dark));
  }

  &--arrow-left-bottom {
    @include arrow(left, bottom, map-fetch($color, ui, dark));
  }

  &--arrow-left-center {
    @include arrow(left, center, map-fetch($color, ui, dark));
  }

  &--arrow-left-top {
    @include arrow(left, top, map-fetch($color, ui, dark));
  }
}
