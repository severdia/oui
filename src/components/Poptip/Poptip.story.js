import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, select } from '@storybook/addon-knobs';

import Poptip from './index.js';
import Button from '../Button';
import Icon from 'react-oui-icons';

const stories = storiesOf('Poptip', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      <div className="flexboxCenter">
        {story()}
      </div>
    </div>
  ));

stories
  .add('Basic poptip', (() => (
    <Poptip
      trigger='click'
      content="This is a sample poptip."
      position="right">
      <Button width="default">
        Default Poptip
      </Button>
    </Poptip>
  )))
  .add('Poptip with children inline', (() => (
    <div
      className="flex flex-align--center line--1">
      <input
        className="text-input width--300 push-half--right"
        type="text"
      />
      <Poptip
        content="Block Origin"
        position="right"
        isInline={ true }>
        <Icon name='ban'/>
      </Poptip>
    </div>
  )))
  .add('Poptip to the right', (() => (
    <Poptip
      trigger='mouseenter'
      content="This is a sample poptip."
      position='right'>
      <Button style="unstyled" width="default">
        <a>Poptip on the right</a>
      </Button>
    </Poptip>
  )))
  .add('Poptip light theme', (() => (
    <Poptip
      trigger='click'
      content="This is a sample poptip."
      theme='light'>
      <Button style="unstyled" width="default">
        <a>Click Poptip, light theme</a>
      </Button>
    </Poptip>
  )))
  .add('Poptip light theme on the right', (() => (
    <Poptip
      trigger='click'
      content="This is a sample poptip."
      theme='light'
      position='right'>
      <Button style="unstyled" width="default">
        <a>Click Poptip, light theme, on the right</a>
      </Button>
    </Poptip>
  )))
  .add('Customizable with knobs', (() => (
    <Poptip
      content={ 'This is a sample poptip.' }
      disable={ boolean('disable', false) }
      isAnimated={ boolean('isAnimated', true) }
      position={ select('position', ['top', 'right', 'bottom', 'left'], 'top') }
      theme={ select('theme', ['dark', 'light', 'transparent'], 'dark') }
      trigger={ select('trigger', ['mouseenter', 'focus', 'click', 'manual'], 'mouseenter') }>
      <Button width="default">
        Customizable Poptip
      </Button>
    </Poptip>
  )))
  .add('Basic poptip with delay', (() => (
    <Poptip
      trigger='mouseenter'
      delay={ 1000 }
      hideDelay={ 5000 }
      content="Ipsa officiis bad-news minus earum a aperiam! Aperiam reiciendis vitae nihil libero et, hic!">
      <Button width="default">
        Default Poptip
      </Button>
    </Poptip>
  )));
