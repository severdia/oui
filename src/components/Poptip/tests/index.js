import React from 'react';
import Poptip from '../index';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Tooltip } from 'react-tippy';

describe('components/Poptip', () => {
  it('renders correctly', () => {
    const component = shallow(<Poptip content="Really interesting info!">Testing Poptip</Poptip>);
    component.props().setBodyDefined(true);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should call the callback onClick', () => {
    const mockCallBack = jest.fn();
    const component = shallow(
      <Poptip content="Really interesting info!"><button onClick={ mockCallBack }>Testing Poptip</button></Poptip>
    );
    component.props().setBodyDefined(true);
    component.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('should call the callback onMouseOver', () => {
    const mockCallBack = jest.fn();
    const component = shallow(
      <Poptip content="Really interesting info!"><button onMouseOver={ mockCallBack }>Testing Poptip</button></Poptip>
    );
    component.props().setBodyDefined(true);
    component.find('button').simulate('mouseover');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('should pass the delay and hideDelay prop to Tooltip', () => {
    const delay = 1000;
    const hideDelay = 2000;
    const component = mount(
      <Poptip content="Really interesting info!" delay={ delay } hideDelay={ hideDelay }>Testing Poptip</Poptip>
    );
    expect(component.find(Tooltip).prop('delay')).toEqual([delay, hideDelay]);
  });
});
