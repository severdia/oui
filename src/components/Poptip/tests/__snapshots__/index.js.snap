// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Poptip renders correctly 1`] = `
<Poptip
  bodyDefined={true}
  content="Really interesting info!"
  delay={0}
  disable={false}
  hideDelay={0}
  isAnimated={true}
  isInline={false}
  position="top"
  setBodyDefined={[Function]}
  theme="dark"
>
  Testing Poptip
</Poptip>
`;
